src/main/scala/org/broadinstitute/hail/expr/AST.scala

```diff
@@ -222,13 +223,34 @@ case object TChar extends Type {
     .map(s => s.substring(0, 1))
 }
 
-abstract class TNumeric extends Type
+object TNumeric {
+  def parseResultType(types: Set[TNumeric]): Type = {
+    if (types(TDouble) || (types(TFloat) && (types(TInt) || types(TLong))))
+      TDouble
+    else if (types(TInt) && types(TLong))
+      TLong
```

Comment:
OK.  I'm suggesting you return a float if there are floats and no doubles.

The conditions in this code seem unnecessarily complex.  What if there are multiple longs but no ints?  It seems like it fails (assert(types.size == 1)).  I'm suggesting roughly:

```
  if (types(TDouble))
    TDouble
  else if (types(TFloat))
    TFloat
  ...
```